source/common/request_source_plugin_impl.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include "envoy/registry/registry.h"
+
+#include "nighthawk/common/request_source_plugin.h"
+
+#include "api/client/options.pb.h"
+#include "api/request_source/request_source_plugin_impl.pb.h"
+
+#include "common/uri_impl.h"
+
+namespace Nighthawk {
+
+/**
+ */
+class DummyRequestSourcePlugin : public RequestSourcePlugin {
```

Comment:
Plz correct me if I'm wrong, it seems DummyRequestSourcePlugin can directly inherit from RequestSource?

Reply:
So, generally speaking, you're right. However, in this case it's made awkward by the fact that I want the create method for the generic case to take in an api parameter. This means that unless I make the constructor for all RequestSourcePlugins take the api parameter, the compiler complains about an unused parameter. I agree that this seems inelegant, but I wasn't sure if there was a better solution for this.

Reply:
nevermind, I was dumb.